src/internal/watch/postgres/impl.go

```diff
@@ -0,0 +1,144 @@
+package postgres
+
+import (
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/lib/pq"
+	"github.com/pachyderm/pachyderm/v2/src/internal/collection"
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pachsql"
+)
+
+type EventType int
+
+const (
+	// EventInsert happens when an item is added
+	EventInsert EventType = iota
+	// EventUpdate happens when an item is updated
+	EventUpdate
+	// EventDelete happens when an item is removed
+	EventDelete
+	// EventError happens when an error occurred
+	EventError
```

Comment:
Note to self, this is probably not needed since a non-nil error value in the event is enough.